builder_limit.go

```diff
@@ -14,28 +14,25 @@ func (b *Builder) limitWriteTo(w Writer) error {
 	if strings.TrimSpace(b.dialect) == "" {
 		return errors.New("field `dialect` must be set up when performing LIMIT, try use `Dialect(dbType)` at first")
 	}
-	if !(b.optype == selectType || b.optype == unionType) {
-		return errors.New("LIMIT operation is limited in SELECT and UNION")
-	}
 
 	if b.limitation != nil {
 		limit := b.limitation
 		if limit.offset < 0 || limit.limitN <= 0 {
 			return errors.New("unexpected offset/limitN")
 		}
-
-		selects := b.selects
 		// erase limit condition
 		b.limitation = nil
-		// flush writer, both buffer & args
 		ow := w.(*BytesWriter)
-		ow.writer.Reset()
-		ow.args = nil
 
 		var final *Builder
 
 		switch strings.ToLower(strings.TrimSpace(b.dialect)) {
 		case ORACLE:
+			// flush writer, both buffer & args
+			ow.writer.Reset()
```

Comment:
We could copy ow to a new Builder and don't reset it for performance.

Reply:
Further more, I want to use Builder compents to build its new functions.  This may cause a performance problem. I'll try to optimize related code for general LIMIT op.